app/template/rest.py

```diff
@@ -245,11 +245,15 @@ def preview_letter_template_by_notification_id(service_id, notification_id, file
                 status_code=500
             )
 
-        content = base64.b64encode(pdf_file).decode('utf-8')
-        overlay = metadata.get("message") == "content-outside-printable-area"
         page_number = page if page else "1"
+        content = base64.b64encode(pdf_file).decode('utf-8')
+        content_outside_printable_area = metadata.get("message") == "content-outside-printable-area"
+
+        show_overlay_for_page = False
+        if content_outside_printable_area and page_number in metadata.get('invalid_pages', '[]'):
+            show_overlay_for_page = True
 
-        if overlay:
+        if show_overlay_for_page or (content_outside_printable_area and file_type == "pdf"):
```

Comment:
Oh, I'm open to suggestions on how to improve all the logic in this method as it's all fairly complex so if you can see a nicer way to reorganise everything then I'd like to hear about it

Reply:
Nice suggestion, I've implemented.

Agreed the whole function is gnarly. I think I'd like to potentially see it refactored so that pdf and pngs are just different routes but something I won't look at doing now.